### PR TITLE
[SYCL] Release specialization constant buffer

### PR DESCRIPTION
**Pull request by HabKaffee**
This PR includes tests for implemented spec const buffer.
Implementation in intel/llvm.
link to PR with implementation: https://github.com/intel/llvm/pull/5167

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,8 @@
 # External/*
 /test-suite-externals
 *.pyc
+build/
+Makefile.config
+config.log
+config.status
+mklib
```
Comment (steffenlarsen) [inline .gitignore]: Was this as mistake? If not then it should be in a separate PR.

Comment (HabKaffee) [reply .gitignore]: I added it just not to upload some build files. If these files are necessary, I'll delete them from .gitignore.

Comment (HabKaffee) [reply .gitignore]: I accidentally put .gitignore to this PR. I'll delete it from PR

Comment (steffenlarsen) [reply .gitignore]: You don't have to delete all of .gitignore. I am not saying the additions to the .gitignore is necessarily a bad idea, just that it should be a separate PR so it can be reviewed independenty of the new test.

Comment (HabKaffee) [reply .gitignore]: OK, I got your point.

Comment (bader) [reply .gitignore]: > You don't have to delete all of .gitignore.
> I got your point.

But you "deleted all of .gitignore". :)

Comment (HabKaffee) [reply .gitignore]: I thought it was a good option not to put it in this PR, because it wouldn't affect on workability at all. :)

Comment (bader) [reply .gitignore]: `it`? What Steffen said is - do not make any changes to `.gitignore` in this PR. Currently your PR removes `.gitignore` file. I don't think we should do this, but if you really want to make changes to this file, please, make them in a separate PR.
Please, recover original content of the `.gitignore` file.

Comment (HabKaffee) [reply .gitignore]: I recovered original `.gitignore` file. Thanks for clarification.

File: SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// REQUIRES: opencl-aot, cpu
+
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64 %S/Inputs/common.cpp -o %t.out \
+// RUN:          -fsycl-dead-args-optimization
+// RUN: env SYCL_PI_TRACE=-1 %CPU_RUN_PLACEHOLDER %t.out | FileCheck %s
+
+#include <cmath>
+#include <sycl/sycl.hpp>
+
+class Kernel1Name;
```
Comment (steffenlarsen) [inline SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp]: This is redundant as you use it as `class Kernel1Name` in `single_task`.

File: SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// REQUIRES: opencl-aot, cpu
+
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64 %S/Inputs/common.cpp -o %t.out \
+// RUN:          -fsycl-dead-args-optimization
+// RUN: env SYCL_PI_TRACE=-1 %CPU_RUN_PLACEHOLDER %t.out | FileCheck %s
+
+#include <cmath>
```
Comment (steffenlarsen) [inline SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp]: Is anything from `cmath` being used?

File: SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+#include <sycl/sycl.hpp>
+
+class Kernel1Name;
+
+const static sycl::specialization_id<int> SpecConst2{42};
```
Comment (steffenlarsen) [inline SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp]: Nit: The numbering in `SpecConst2` seems a bit unnecessary as it is the only specialization constant in this test. Same goes for `Kernel1Name`.

**Review by steffenlarsen — COMMENTED** [review]
Some minor comments, but generally it looks good.

File: .gitignore (hunk 1 above)
Comment (steffenlarsen) [reply .gitignore]: Thank you!

File: SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp (hunk 3 above)
Comment (HabKaffee) [reply SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp]: No, I'll remove it, because it is unnecessary.

File: SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp (hunk 4 above)
Comment (HabKaffee) [reply SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp]: You're right, the numbering is unnecessary in this test.

File: SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp (hunk 2 above)
Comment (HabKaffee) [reply SYCL/SpecConstants/2020/non_native/SpecConstBuffer.cpp]: Yes, I've already removed it.

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by bader** [comment]
@kbobrovs, ping.

**Comment by HabKaffee** [comment]
Nothing new was added by last commit.

**Comment by HabKaffee** [comment]
`Jenkins/pre-ci-linux` failed due to fail in `SYCL :: DiscardEvents/invalid_event.cpp`. My test does not affect this test at all.

**Review by vladimirlaz — APPROVED** [review]
